internal/provider/incident_alert_attribute_resource.go

```diff
@@ -132,10 +133,14 @@ func (r *IncidentAlertAttributeResource) Read(ctx context.Context, req resource.
 	}
 
 	result, err := r.client.AlertAttributesV2ShowWithResponse(ctx, data.ID.ValueString())
-	if err == nil && result.StatusCode() >= 400 {
-		err = fmt.Errorf(string(result.Body))
-	}
 	if err != nil {
+		// Check if error message contains any indication of a 404 not found
+		errStr := err.Error()
+		if strings.Contains(errStr, "404") || strings.Contains(errStr, "not found") || strings.Contains(errStr, "resource_not_found") {
```

Comment:
Rather than using string-matching, could we use `errors.As` to turn this into a `client.HTTPError` and check the `.StatusCode` on that?

```suggestion
		httpErr := client.HTTPError{}
    if errors.As(err, &httpErr) && httpErr.StatusCode == 404 {
```